client.js

```diff
@@ -26,5 +26,30 @@ on('onClientResourceStart', (resource) => {
                 }
             }, 1500);
         }
+        console.log('[snow] snow enabled');
+    }
+});
+
+on('onResourceStop', (resource) => {
+    if (resource === 'snow') {
+        ClearOverrideWeather();
+        SetSnowLevel(-1.0);
+
+        // Calling this native seems to crash the script no matter how it's used. It's only used 4 times in the whole game, it could be just buggered.
+        //UnloadCloudHat('Snowy 01', 1); 
+
+        SetForcePedFootstepsTracks(false);
+        SetForceVehicleTrails(false);
+        SetRainLevel(0.0);
+        RemoveNamedPtfxAsset('core_snow');
+
+        if (GetConvar('snow_spawnTree', 'true').match(/^("true"|true)$/i)) {
+            if (HasModelLoaded(xmasTreeHash)) {
```

Comment:
is the model loaded check needed? surely if the engine at some point unloads the model it'll fail to remove the entity?